src/main/kotlin/net/fabricmc/language/kotlin/KotlinLanguageAdapter.kt

```diff
@@ -16,24 +16,15 @@
 
 package net.fabricmc.language.kotlin
 
-
 import net.fabricmc.loader.language.ILanguageAdapter
 import org.apache.logging.log4j.LogManager
 
 class KotlinLanguageAdapter : ILanguageAdapter {
 
     private val logger = LogManager.getFormatterLogger("KotlinLanguageAdapter")
 
-    override fun createInstance(clazz: Class<*>): Any {
-        return try {
-            val instanceField = clazz.getField("INSTANCE")
-            val instance = instanceField.get(null) ?: throw NullPointerException()
-            logger.debug("Found INSTANCE field for ${clazz.name}")
-            instance
-        } catch (e: Exception) {
-            logger.debug("Unable to find INSTANCE field for ${clazz.name}, constructing new instance")
-            clazz.newInstance()
-        }
+    override fun createInstance(clazz: Class<*>): Any = clazz.kotlin.objectInstance ?: run {
+        logger.debug("Unable to find INSTANCE field for ${clazz.name}, constructing new instance")
```

Comment:
This should be `Kotlin object instance`, not `INSTANCE field`

Reply:
fix